internal/command/run.go

```diff
@@ -41,3 +45,64 @@ func RunWithExtraEnvs(executable string, args []string, extraEnvs []string) (out
 	errBytes = stderr.Bytes()
 	return outBytes, errBytes, err
 }
+
+// RunWithStreamingOutput invokes exe with given args and env. Stdout and stderr
+// are streamed to outWriter and errWriter, respectively. If exe errors with an
+// exit code equal to failureExitCode, no error is returned to the caller,
+// but success is false. If exe does not error, success is true.
+func RunWithStreamingOutput(exe string, args []string, env []string, outWriter, errWriter io.Writer, failureExitCode int) (success bool, err error) {
+	return runWithStreamingOutput(exe, args, env, outWriter, errWriter, failureExitCode, false)
+}
+
+// RunWithStreamingOutput invokes exe with given args and env. Stdout and stderr
+// are streamed to outWriter and errWriter, respectively. If exe errors with an
+// exit code equal to failureExitCode, no error is returned to the caller,
+// but success is false. If exe does not error, success is true.
+// RunWithStreamingOutputReversed is identical to RunWithStreamingOutput,
+// except for streaming stderr before stdout.
+func RunWithStreamingOutputReversed(exe string, args []string, env []string, outWriter, errWriter io.Writer, failureExitCode int) (success bool, err error) {
+	return runWithStreamingOutput(exe, args, env, outWriter, errWriter, failureExitCode, true)
+}
+
+// runWithStreamingOutput is the private implementation of RunWithStreamingOutput(Reversed).
+func runWithStreamingOutput(exe string, args []string, env []string, outWriter, errWriter io.Writer, failureExitCode int, stderrFirst bool) (success bool, err error) {
+	cmd := exec.Command(exe, args...)
+	cmd.Env = append(os.Environ(), env...)
+	cmdStderr, err := cmd.StderrPipe()
+	if err != nil {
```

Comment:
in the case we use `RunWithStreamingOutputReversed ` shouldn't we say the other way around regarding what pipe is connected?

Reply:
I guess not because we want to behave it like that explicitly?

Reply:
Yea we want to print it in "reversed" order because printing the summary is more intuitive - it would be weird to print the summary before the logs.